### ISSUE TITLE
[Bug] Application freezes on trying to download repository metadata

### ISSUE THREAD

**Issue by craftersmine** [KSP-CKAN/CKAN#3814]
Background
----------
* Operating System: Windows 11 x64 22H2 Release Candidate
* CKAN Version: 1.31.3
* KSP Version: 1.12.5.3190
* .NET: Framework 4.8, 6.0 (Runtime, SDK), 7.0 (Runtime, SDK)

**Have you made any manual changes to your GameData folder (i.e., not via CKAN)?**
No, only backed up them and removed (both in %localappdata% and in %ksp_install_dir%), which didn't helped to resolve the issue

Problem
-------
* CKAN completely frozen when launched and trying to download a mods repository from GitHub

**Steps to reproduce**
1. Launch the CKAN
2. Wait for `Downloading "%ckan-repo%"` message to appear.
3. CKAN completely frozen, not responding, no CPU, RAM or Disk/Network I/O.

**Expected behavior**
Window should show either successful download of repository or fail to download and fallback to installed-only mod list

**Screenshots (if applicable)**
![image](https://user-images.githubusercontent.com/11241645/224795963-7c4aca59-df02-4e8a-a5b5-4788c720b0c2.png)

**CKAN error code (if applicable):**
* Doesn't crash neither generate a stacktrace or logs (or console output)

**Comment by HebaruSan** [comment]
Duplicate of #3802

**Comment by craftersmine** [comment]
> Duplicate of #3802

Ok, yeah, that fixed, I've caught a malware not that long ago, and was confused why I can't open a GitHub page (and a bunch of others, which were just redirecting me to dns.google page) and I checked `hosts` file right now and it was bloated with DNS redirections, thanks. But CKAN was showing that it couldn't download a mod repository due to SSL certificate issues before, and then started to freeze when I launch it

**Comment by HebaruSan** [comment]
So is there a malware in the wild now that adds a bunch of junk to your hosts file? Interesting.

**Comment by craftersmine** [comment]
I think I caught a miner, but I've managed to removed all it's executables and directories, but apparently, it wrote a bunch of DNS redirections for sites, such as antivirus sites, IT help sites, etc, but I didn't noticed this and was confused why some sites domains were redirected to `dns.google` page. And that's how I got to issues with CKAN, it tries to download TAR archive from `https://github.com/KSP-CKAN/CKAN-meta/archive/master.tar.gz`, but it redirects to domain with `https://codeload.github.com` which was added to `hosts` as `8.8.8.8 codeload.github.com`. Interesting is that when I first launched CKAN, it was showing the SSL/TLS connection cannot be established error. I've downloaded repository archive with my phone and added it as local file using `file:///D:\path_to_archive\master.tar.gz` and removed `default` repository. It worked, until it doesn't, something borked due to this, and it started freezing when I launch CKAN, removed directories (copied them and removed) and after relaunch, it thought it was first start, but it still froze on trying to download a repository.
